### PR TITLE
Create dynamic referencing of classes

### PR DESCRIPTION
**Pull request by dganev-cb**
## Pull request checklist

Please check if your PR fulfils the following requirements:
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?
3121

## Pull Request Description
I think it will be a good addition to our ORM-like SDK and since its widely used in the Python community in frameworks such as Django's ORM, Flask's Blueprints and etc. 

And when I first interacted with the SDK and saw how it works I thought it would have that functionality, simply sharing this to give you my perspective of how I saw the SDK at first. Definitely needs more research from our clients since they are not heavy python users.

Referer to the other argumentation in the ticket: 3121 

## Does this introduce a breaking change?

- [x] No

## How Has This Been Tested?
Currently the tests has been partially done for the Alerts only. The testing must cover all of our queryable models.

### CODE REVIEW

**Comment by dganev-cb** [comment]
I've added a testing example of how it might be tested. @abowersox-cb can you check it out and tell me what do you think about it. I've decided that `__qualname__` would be a better alternative than just `__name__` since we have nested classes.

**Comment by dganev-cb** [comment]
> I've added a testing example of how it might be tested. @abowersox-cb can you check it out and tell me what do you think about it. I've decided that `__qualname__` would be a better alternative than just `__name__` since we have nested classes.

Actually now thinking about it I should probably only test the `select_class_instance` function since the rest of the logic remains the same in the `select` gonna update the unit tests.

**Comment by abowersox-cb** [comment]
Can you fix the flake8 errors? I want to make sure the new unit tests run before I approve this, because I'm still not sure how you're dealing with the module-loading issue.

**Comment by dganev-cb** [comment]
> Can you fix the flake8 errors? I want to make sure the new unit tests run before I approve this, because I'm still not sure how you're dealing with the module-loading issue.

I will do that asap, I am still figuring out why those modules are not loading  `SensorKit`, `ComputeResource`.

**Comment by dganev-cb** [comment]
@abowersox-cb @avanbrunt-cb It turns out that, the `SensorKit, ComputeResource` are the only 2 modules that are not loading. I've looked for other solutions and currently, I have 2.

1. We import them in the `__init__.py` level and with that we can have it loaded.
2. Go with an entirely different solution by using a `dict` with keys as the classes and values with `import_module` function. 

I don't like either of those and I've asked around to see if I can find some better solutions. I will put this into a stall for now.

**Comment by dganev-cb** [comment]
@abowersox-cb @avanbrunt-cb I've added a list variable with `import_module` function just to make sure that everytime we call `select_class_instance` all of our modules are imported. This way we won't need to address an addition or a change whenever we create new models. 

I've also changed the name of the Event in the Endpoint Standard to `EndPointStandardEvent` since that I am not sure if that was the model that is going to be deprecated I will leave this at this stage. Maybe a more appropriate name or an eventual usage of a `@deprecated` decorator can be used.

**Comment by dganev-cb** [comment]
Since I found new information over this feature I am updating it.

We can use the `pkgutil.walk_packages` to iterate over all the modules located in the cbc_sdk and load them. 
(https://docs.python.org/3/library/pkgutil.html#pkgutil.walk_packages) 

The only thing that is not tested is the `Platform.Event` because it is going to be deprecated, it clashes with the `endpoint_standard.Event`. 

Tell me what do you think @abowersox-cb @emitreva-cb @avanbrunt-cb

**Review by avanbrunt-cb — COMMENTED** [review]
We should not merge this until we remove the event code which should be done by the end of January given that functionality will no longer be available in the CBC

File: requirements.txt
Comment (avanbrunt-cb) [inline requirements.txt:15]: I don't see where this is being used. We need to be careful adding too many new packages.

File: src/tests/unit/base/test_queries.py
Comment (avanbrunt-cb) [inline src/tests/unit/base/test_queries.py:123]: Using the following code does not reference the correct Event model
```
cb.select('Event', '7DESJ9GN-0040bdb5-000009e0-00000000-1d82fe3b49a5b32')
```

instead it matches to the following
```
<cbc_sdk.endpoint_standard.base.Event: id 7DESJ9GN-0040bdb5-000009e0-00000000-1d82fe3b49a5b32> @ https://defense.conferdeploy.net
```

Was expecting this instead
```
<cbc_sdk.platform.events.Event: id 7DESJ9GN-0040bdb5-000009e0-00000000-1d82fe3b49a5b32> @ https://defense.conferdeploy.net
```

**Review by avanbrunt-cb — CHANGES_REQUESTED** [review]

**Comment by dganev-cb** [comment]
@avanbrunt-cb I cannot replicate that, and this shouldn't be possible since the `Event` that you are referring to doesn't extend the `NewBaseModel`.

**Review by avanbrunt-cb — APPROVED** [review]
I've modified the test file as the `_test` name caused the tests to be ignored. I've also enhanced the event test to include the other approach of specifying the process_guid in order to ensure both mechanisms import the platform models. Lastly, the deprecation package was still included in the requirements.txt file which was not being utilized.

I am comfortable with the changes now after further testing and ensuring I had the latest changes

**Review by abowersox-cb — APPROVED** [review]
Good work cleaning this up, @avanbrunt-cb.

**Pull request merged**
